README.md

```diff
@@ -144,81 +144,146 @@ The old repo is no longer the recommended approach for running and extending the
 
-### Option 1: Run the Mech as a standalone agent
+For the variable `SKILL_MECH_ABCI_MODELS_BENCHMARK_TOOL_ARGS_LOG_DIR`, create a folder `tmp` and copy its absolute path, and set the variable `SKILL_MECH_ABCI_MODELS_BENCHMARK_TOOL_ARGS_LOG_DIR` to this path.
 
-1. Ensure you have a file with a private key (`ethereum_private_key.txt`). You can generate a new private key file using the Open Autonomy CLI:
+If you want to run a legacy Mech, the `SKILL_TASK_EXECUTION_MODELS_PARAMS_ARGS_MECH_MARKETPLACE_ADDRESS` is optional. Otherwise this variable needs to be defined. 
+Furthermore, in the variable `SKILL_TASK_EXECUTION_MODELS_PARAMS_ARGS_MECH_TO_CONFIG`, the value corresponding to the key `is_marketplace_mech` should be set to true.
+Note that even in this case, the Mech won't run without changing also the other variables. 
 
-   ```bash
-   autonomy generate-key ethereum 
-   ```
+Ensure that the variable `SKILL_MECH_ABCI_MODELS_PARAMS_ARGS_SETUP_ALL_PARTICIPANTS` in the file `.agentenv` contains the same agent instance address as in `keys.json`.
 
-2. From one terminal, run the agent:
+Once you are done modifying the `.1env` file, run the following: 
 
-    ```bash
-    bash run_agent.sh
-    ```
+```bash
+# Source the env file
+source .1env
+```
 
-3. From another terminal, run the Tendermint node:
+Finally, in the file `run_service.sh`, change the line `autonomy deploy build -ltm` to `autonomy deploy build -ltm -n k`,
+where k is replaced by the number of agents in your service. For instance for 1 agent: `autonomy deploy build -ltm -n 1`.
 
-    ```bash
-    bash run_tm.sh
-    ```
+The rest of the common environment variables are present in the [service.yaml](https://github.com/valory-xyz/mech/blob/main/packages/valory/services/mech/service.yaml), which are customizable too.
 
-### Option 2: Run the Mech as an agent service
+#### Running the service 
 
-1. Ensure you have a file with the agent address and private key (`keys.json`). You can generate a new private key file using the Open Autonomy CLI:
 
-    ```bash
-    autonomy generate-key ethereum -n 1
+1. Run the service:
+
     ```
+    poetry shell
+    poetry install
+    bash run_service.sh
+    ```
+
+2. Build the docker container by running the following:
+
+```
+cd mech
+build_dir=$(ls -d abci_build_????/ 2>/dev/null || echo "abci_build")
+autonomy deploy run --build-dir $build_dir
+```
+
+
+3. In a separate terminal, run the following to see the logs:
+
+```
+container = $(basename "$(ls -d ../*/abci_build_????/ 2>/dev/null | head -n 1)" | sed -E 's/^abci_build_//')
+container = "mech${container}_abci_0"
+docker logs -f container
+```
```

Comment:
Let's specify that this is for the first agent's logs as the user might be running this as a multi-agent service.

Reply:
@Sfgangloff update the other pr accordingly